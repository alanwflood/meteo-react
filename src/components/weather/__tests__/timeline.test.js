import React from "react";
import { render } from "@testing-library/react";
import WeatherTimeline from "../timeline";
import { getUnixTime, addHours } from "date-fns";

describe("Weather Timeline", () => {
  // 1pm @ 27th September 2010
  const date = new Date(2010, 8, 27, 13, 0, 0);
  const mockWeatherData = {
    times: [
      getUnixTime(date),
      getUnixTime(addHours(date, 3)),
      getUnixTime(addHours(date, 6)),
    ],
    icons: ["01d", "01d", "01d"],
    descriptions: ["Sunny", "Clear", "Delightful"],
  };

  it("Renders correctly", () => {
    const { container } = render(
      <WeatherTimeline weatherData={mockWeatherData} />
    );
    expect(container).toMatchSnapshot();
  });

  it("Shows correct times in timeline", () => {
    const { getByText } = render(
      <WeatherTimeline weatherData={mockWeatherData} />
    );
    expect(getByText("1:00PM")).toBeTruthy();
    expect(getByText("4:00PM")).toBeTruthy();
    expect(getByText("7:00PM")).toBeTruthy();
  });
});
