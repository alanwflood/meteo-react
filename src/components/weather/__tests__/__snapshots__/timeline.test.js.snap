// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather Timeline Renders correctly 1`] = `
<div>
  <section>
    <h3
      style="text-align: center;"
    >
      Timeline
    </h3>
    <div
      class="timeline"
    >
      <div
        class="weather-entry"
      >
        <div
          class="time"
        >
          1:00PM
        </div>
        <div
          class="description"
        >
          <div
            class="icon weather-icon"
          >
            <test-file-stub />
          </div>
        </div>
      </div>
      <div
        class="weather-entry"
      >
        <div
          class="time"
        >
          4:00PM
        </div>
        <div
          class="description"
        >
          <div
            class="icon weather-icon"
          >
            <test-file-stub />
          </div>
        </div>
      </div>
      <div
        class="weather-entry"
      >
        <div
          class="time"
        >
          7:00PM
        </div>
        <div
          class="description"
        >
          <div
            class="icon weather-icon"
          >
            <test-file-stub />
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
